web/client-api/src/main/java/io/deephaven/web/client/api/i18n/JsTimeZone.java

```diff
@@ -26,108 +29,525 @@ private static JsTimeZone createTimeZone(String tzCode) {
+        addJsonMapping("Asia/Kolkata", constants.asiaCalcutta());
+        addJsonMapping("IN", constants.asiaCalcutta());
+        addJsonMapping("IST", constants.asiaCalcutta());
+        addJsonMapping("TW", constants.asiaTaipei());
+        addJsonMapping("NL", constants.europeAmsterdam());
+        addJsonMapping("CE", constants.europeBerlin());
+        addJsonMapping("CES", constants.europeBerlin());
+        addJsonMapping("CEST", constants.europeBerlin());
+        addJsonMapping("LON", constants.europeLondon());
+        addJsonMapping("BST", constants.europeLondon());
+        addJsonMapping("CH", constants.europeZurich());
+        addJsonMapping("BT", constants.americaSaoPaulo());
+        addJsonMapping("BRST", constants.americaSaoPaulo());
+        addJsonMapping("BRT", constants.americaSaoPaulo());
+        addJsonMapping("NF", constants.americaStJohns());
+        addJsonMapping("NST", constants.americaStJohns());
+        addJsonMapping("NDT", constants.americaStJohns());
+        addJsonMapping("AT", constants.americaHalifax());
+        addJsonMapping("AST", constants.americaHalifax());
+        addJsonMapping("ADT", constants.americaHalifax());
+        addJsonMapping("ET", constants.americaNewYork());
+        addJsonMapping("EST", constants.americaNewYork());
+        addJsonMapping("EDT", constants.americaNewYork());
+        addJsonMapping("MN", constants.americaChicago());
+        addJsonMapping("CT", constants.americaChicago());
+        addJsonMapping("CST", constants.americaChicago());
+        addJsonMapping("CDT", constants.americaChicago());
+        addJsonMapping("MT", constants.americaDenver());
+        addJsonMapping("MST", constants.americaDenver());
+        addJsonMapping("MDT", constants.americaDenver());
+        addJsonMapping("PT", constants.americaLosAngeles());
+        addJsonMapping("PST", constants.americaLosAngeles());
+        addJsonMapping("PDT", constants.americaLosAngeles());
+        addJsonMapping("AL", constants.americaAnchorage());
+        addJsonMapping("AKST", constants.americaAnchorage());
+        addJsonMapping("AKDT", constants.americaAnchorage());
+        addJsonMapping("HI", constants.pacificHonolulu());
+        addJsonMapping("HST", constants.pacificHonolulu());
+        addJsonMapping("HDT", constants.pacificHonolulu());
+        addJsonMapping("SYD", constants.australiaSydney());
+        addJsonMapping("AEST", constants.australiaSydney());
+        addJsonMapping("AEDT", constants.australiaSydney());
+        addJsonMapping("MOS", constants.europeMoscow());
+        addJsonMapping("SHG", constants.asiaShanghai());
+    }
+
+    /**
+     * Create a time zone from the JSON provided and add to the map with it's ID
+     * 
+     * @param json The JSON encoded time zone info
+     */
+    private static void addJsonTimeZone(String json) {
+        TimeZone tz = TimeZone.createTimeZone(json);
+        addMapping(tz.getID(), json);
```

Comment:
Hang on so we pre-parse all of them up front, then throw the parsed content away? Wouldnt it make more sense to keep it parsed, since we've already done the work?

Reply:
Yea I was throwing it away because when I was doing performance testing, it looked like the heap was growing ~1MB... but running the tests again I don't think that's an accurate measurement and it doesn't seem to be a significant amount. However, the bundle size of dh-core.js is only 1.9MB instead of 2.2MB if we just store the TimeZone object instead of the JSON string, so I'll switch it to that. Also makes the GMT handling a bit cleaner.